cachelib/allocator/CacheAllocator-inl.h

```diff
@@ -1203,67 +1276,81 @@ bool CacheAllocator<CacheTrait>::moveRegularItem(Item& oldItem,
     XDCHECK(!oldItem.hasChainedItem());
     XDCHECK(newItemHdl->hasChainedItem());
   }
+  
+  auto predicate = [&](const Item& item){
+    // we rely on moving flag being set (it should block all readers)
+    XDCHECK_EQ(item.getRefCount(),0);
+    XDCHECK(item.isMoving());
+    return item.isMoving();
```

Comment:
we can probably just return true here, right? no need to check the isMoving in Release

Reply:
Yeah. My strategy was to extract code from develop as is and use this PR to discuss excessive `XDCHECK`. I believe Daniel added them when investigated data race issues.
In that particular case there are two possible simplifications:
- remove checks and just `return item.isMoving();`
- hardcode to always return `true` and keep checks.